docs(): improved keyboard/focus management

marcysutton opened the pull request:
With this change, focus is only sent to the main content area on click of a nav link. There is also a "Skip to Content" link that allows users to skip past the sidebar navigation.

![Skip to content link](https://cloud.githubusercontent.com/assets/1045233/6428724/8ce7f67a-bf60-11e4-936d-c25056ba7458.png)


Closes #1723

ThomasBurleson commented on docs/app/css/style.css:
```diff
@@ -51,6 +51,31 @@ ul li:first-child {
+  display: block;
+  font-size: 1.25em;
+  margin: 0.5em;
+  opacity: 0;
```
So by using opacity:0, it is not visible... but still rendered. 
Would `display:none;` be a better choice?

marcysutton replied:
The pattern for skip links is to have them always render but show visibly on focus. That way they are in the page for screen reader users at all times.

ThomasBurleson commented on docs/app/js/app.js:
```diff
@@ -287,6 +287,10 @@ function(COMPONENTS, DEMOS, PAGES, $location, $rootScope) {
       };
+
+      $scope.itemClicked = function() {
+        controller.navItemClicked = true;
```
Instead of `navItemClicked`, I recommend that we call this variable `autoFocusContent`; since we do not care **how** the value was set to true, but rather **why** the variable is used.

marcysutton replied:
Sounds good!

ThomasBurleson commented on docs/app/partials/menu-link.tmpl.html:
```diff
@@ -1,5 +1,5 @@
 <md-button ng-class="{'active' : isSelected()}"
-  ng-href="#{{section.url}}">
+  ng-href="#{{section.url}}" ng-click="itemClicked()">
```
I assume that `ng-click` has higher priority that `ng-href`?

marcysutton replied:
We can name it whatever we want. The purpose of `ng-click` here was to provide a hook differentiating link clicks from route changes. There wasn't a way to do that with `ng-href`.

marcysutton commented:
@ThomasBurleson I incorporated your changes and fixed a horizontal scrollbar issue in the sidenav. Going to merge to master and we can tweak it later if needed.
